src/NaNMath.jl

```diff
@@ -61,6 +61,45 @@ function sum(x::AbstractArray{T}) where T<:AbstractFloat
     return result
 end
 
+"""
+NaNMath.median(A)
+
+##### Args:
+* `A`: An array of floating point numbers
+
+##### Returns:
+*   Returns the median of all elements in the array, ignoring NaN's.
+    Returns NaN for an empty array or array containing NaNs only.
+
+##### Examples:
+```julia
+using NaNMath
+NaNMath.median([1., 2., 3., NaN]) # result: 2.
+NaNMath.median([1., 2., NaN]) # result: 1.5
+NaNMath.median([NaN]) # result: NaN
+```
+"""
+median(x::AbstractArray{<:AbstractFloat}) = median(collect(Iterators.flatten(x)))
+
+function median(x::AbstractVector{<:AbstractFloat})
+
+    x = sort(filter(!isnan, x))
+
+    n = length(x)
+    if n == 0
+        return NaN
```

Comment:
This function is not type stable if the element type isn't Float64. See how the other functions deal with NaN for different floating-point types. (Please add tests with Float32 also.)

Reply:
ah, good spot. is this sufficient to fix it?
```suggestion
        return convert(eltype(x), NaN)
```